### PR TITLE
Thivya/code update

### PR DESCRIPTION
**Pull request by thivya-deriv**
Desc: This is to update the DBOT code by following the cypress best practice.
Card: https://app.clickup.com/t/86byfwpt5

image

### CODE REVIEW

**Review by markodonnell-deriv — APPROVED** [review]
Looks good

File: cypress/support/commands/dBot.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Cypress.Commands.add('c_switchToDemoBot', () => {
+  cy.findByTestId('dt_acc_info').click()
+  cy.xpath("//li[text()='Demo']").click()
+  cy.xpath("//span[@class='acc-switcher__id']/span[text()='Demo']").click()
```
Comment (abuzar-deriv) [inline cypress/support/commands/dBot.js:4]:
```suggestion
  cy.findByText('Demo').click()
  cy.findAllByText('Demo').eq(1).click()
```

File: cypress/support/commands/dBot.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+Cypress.Commands.add('c_runBot', () => {
+  cy.get('.bot-dashboard.bot').should('be.visible')
+  cy.findAllByRole('button', { name: 'Run' }).click()
+})
+
+Cypress.Commands.add('c_stopBot', (waitduration = undefined) => {
+  if (waitduration) {
+    cy.wait(waitduration)
+  }
+  cy.findAllByRole('button', { name: 'Stop' }).click()
```
Comment (abuzar-deriv) [inline cypress/support/commands/dBot.js:17]: If it's finding just one element, please replace it with findByRole. If it's finding more than one, please specify the element index.

File: cypress/support/commands/dBot.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+})
+
+Cypress.Commands.add('c_setBlockDuration', () => {
+  cy.xpath(
+    "(//*[@class='blocklyText' and text()='Duration:']/..)/*[@class='blocklyEditableText']"
+  ).click()
+  cy.findByRole('menuitemcheckbox', { name: 'Ticks' }).click()
+  cy.findByRole('textbox').nth(1).type('2')
```
Comment (abuzar-deriv) [inline cypress/support/commands/dBot.js:25]: Does this work? I believe cy supports eq().

**Review by abuzar-deriv — APPROVED** [review]
LGTM.

**Pull request merged**